packages/back-end/src/integrations/SqlIntegration.ts

```diff
@@ -167,12 +167,21 @@ export default abstract class SqlIntegration
     WITH
       ${experimentQueries
         .map((q, i) => {
+          const hasNameCol = q.hasNameCol || false;
+          const experimentNameCol = hasNameCol
+            ? "experiment_name"
+            : "experiment_id";
+          const variationNameCol = hasNameCol
+            ? "variation_name"
+            : "variation_id";
           return `
         __exposures${i} as (
           SELECT 
-            ${this.castToString(`'${q.id}'`)} as exposure_query, 
+            ${this.castToString(`'${q.id}'`)} as exposure_query,
             experiment_id,
+            MIN(${experimentNameCol}) as experiment_name,
             variation_id,
+            MIN(${variationNameCol}) as variation_name,
```

Comment:
If you aren't using name columns, the query will select the id column both directly and with an aggregation, which I think might throw an error in some database systems:
```sql
SELECT
  variation_id,
  MIN(variation_id) as variation_name
FROM ...
GROUP BY
  variation_id
```

Instead, I would change this logic to:

```
${hasNameCol ? "MIN(variation_name)" : "variation_id"} as variation_name,
```

(same goes for experiment_name)

The end result will be either `MIN(variation_name) as variation_name,` or `variation_id as variation_name,`

Reply:
Resolved by new changes